src/Fixer/Operator/ConcatSpaceFixer.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Operator;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\ConfigurationException\InvalidFixerConfigurationException;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ * @author SpacePossum
+ */
+final class ConcatSpaceFixer extends AbstractFixer
+{
+    private $fixCallback;
+
+    /**
+     * Configuration must have one element 'spacing' with value 'none' (default) or 'one'.
+     *
+     * @param array $configuration
+     */
+    public function configure(array $configuration = null)
+    {
+        if (null === $configuration) {
+            $this->fixCallback = 'fixConcatenationToNoSpace';
+
+            return;
+        }
+
+        if (!array_key_exists('spacing', $configuration)) {
+            throw new InvalidFixerConfigurationException($this->getName(), 'Missing "spacing" configuration.');
+        }
+
+        switch ($configuration['spacing']) {
+            case 'one':
+                $this->fixCallback = 'fixConcatenationToSingleSpace';
+
+                break;
+            case 'none':
+                $this->fixCallback = 'fixConcatenationToNoSpace';
```

Comment:
I noticed, as some SCA will give the `unused private method`-warning. 
While I don't care that much about it I was looking into the PHP internals what the cost is for a dynamic name method call vs. a defined one, but I haven't concluded that little trip down the internals road ;)
I did a few fixers this way and we could stop doing that and update those already accepted at some point.
There are than 2 other ways to do this:
1 - check the configuration each time a token of interest is found to see what method should be called to fix it, with the cost that the answer to that will always be same during execution
2 - within the configuration method create an private interfaced object what will be called during fixing that does the fixing. This will be at the cost of creating more classes/code which means more to maintain.

I've no strong feeling about any of the 3 solutions, but the solution `1` feels like the least optimal.